plugins/zoom-to-fit/src/index.js

```diff
@@ -0,0 +1,251 @@
+    return new Blockly.utils.Rect(
+        this.top_, this.top_ + this.HEIGHT_,
+        this.left_, this.left_ + this.WIDTH_);
+  }
+
+  /**
+   * Positions the zoom-to-fit control.
+   * It is positioned in the opposite corner to the corner the
+   * categories/toolbox starts at.
+   * @param {!Blockly.MetricsManager.UiMetrics} metrics The workspace metrics.
+   * @param {!Array<!Blockly.utils.Rect>} savedPositions List of rectangles that
+   *     are already on the workspace.
+   */
+  position(metrics, savedPositions) {
+    if (!this.initialized_) {
+      return;
+    }
+    const hasVerticalScrollbars = this.workspace_.scrollbar &&
+        this.workspace_.scrollbar.canScrollHorizontally();
+    const hasHorizontalScrollbars = this.workspace_.scrollbar &&
+        this.workspace_.scrollbar.canScrollVertically();
+
+    if (metrics.toolboxMetrics.position === Blockly.TOOLBOX_AT_LEFT ||
+        (this.workspace_.horizontalLayout && !this.workspace_.RTL)) {
+      // Right corner placement.
+      this.left_ = metrics.absoluteMetrics.left + metrics.viewMetrics.width -
+          this.WIDTH_ - this.MARGIN_SIDE_;
+      if (hasVerticalScrollbars && !this.workspace_.RTL) {
+        this.left_ -= Blockly.Scrollbar.scrollbarThickness;
+      }
+    } else {
+      // Left corner placement.
+      this.left_ = this.MARGIN_SIDE_;
+      if (hasVerticalScrollbars && this.workspace_.RTL) {
+        this.left_ += Blockly.Scrollbar.scrollbarThickness;
+      }
+    }
+
+    // Upper corner placement
+    let minTop = metrics.absoluteMetrics.top + this.MARGIN_BOTTOM_;
+    if (hasHorizontalScrollbars) {
+      minTop += Blockly.Scrollbar.scrollbarThickness;
+    }
+
+    // Bottom corner placement
+    const maxTop = metrics.absoluteMetrics.top + metrics.viewMetrics.height -
+        this.HEIGHT_ - this.MARGIN_BOTTOM_;
+    if (hasHorizontalScrollbars) {
+      minTop -= Blockly.Scrollbar.scrollbarThickness;
+    }
+    const placeBottom =
+        metrics.toolboxMetrics.position !== Blockly.TOOLBOX_AT_BOTTOM;
+    this.top_ = placeBottom ? maxTop : minTop;
+
+    // Check for collision and bump if needed.
+    let boundingRect = this.getBoundingRectangle();
+    for (let i = 0, otherEl; (otherEl = savedPositions[i]); i++) {
+      if (boundingRect.intersects(otherEl)) {
+        if (placeBottom) {
+          // Bump up
+          this.top_ = otherEl.top - this.HEIGHT_ - this.MARGIN_BOTTOM_;
+        } else {
+          this.top_ = otherEl.bottom + this.MARGIN_BOTTOM_;
+        }
+        // Recheck other savedPositions
+        boundingRect = this.getBoundingRectangle();
+        i = -1;
+      }
+    }
+    // Clamp top value within valid range.
+    this.top_ = Blockly.utils.math.clamp(minTop, this.top_, maxTop);
+
+    this.svgGroup_.setAttribute('transform',
+        'translate(' + this.left_ + ',' + this.top_ + ')');
+  }
+}
+
+/**
+ * Base64 encoded data uri for zoom to fit  icon.
+ * @type {string}
+ */
+const ZOOM_TO_FIT_SVG_DATAURI =
```

Comment:
Why do you have both a datauri and an svg file?

Reply:
The svg file is committed so that we have a copy of the original svg that the datauri is generated from.
This is the same thing we have done in WorkspaceSearch